.github/workflows/enhanced-ci.yml

```
@@ -0,0 +1,482 @@
+            packages/*/dist
+            apps/*/dist
+            build.log
+          retention-days: 1
+
+  # Performance tests (conditional)
+  performance:
+    name: Performance Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 45
+    needs: [setup, build]
+    if: needs.preflight.outputs.should-run-performance == 'true' || env.PERFORMANCE_TESTS == 'true'
```

actionlint (1.7.8)

property "preflight" is not defined in object type {build: {outputs: {}; result: string}; setup: {outputs: {cache-hit: string}; result: string}}

(expression)